detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UseCheckOrErrorSpec.kt

```diff
@@ -94,5 +101,17 @@ class UseCheckOrErrorSpec : Spek({
             val code = """fun doThrow() { throw IllegalStateException("message") }"""
             assertThat(subject.lint(code)).hasSourceLocation(1, 17)
         }
+
+        it("does not report an issue if the exception thrown has a non-String argument") {
+            val code = """
+                fun test(throwable: Throwable) {
+                    when(throwable) {
+                        is NumberFormatException -> println("a")
+                        else -> throw IllegalStateException(throwable)
+                    }
+                }
+            """.trimIndent()
```

Comment:
It's not necessary to call `.trimIndent()` again, since `lint()` already does it for us.

```suggestion
            """
```